### ISSUE TITLE
`no-restricted-globals` is triggered invalid for interface member

### ISSUE THREAD

**Issue by roydukkey** [eslint/eslint#14947]
**Tell us about your environment**

* **ESLint Version:** ^7.26.0
* **Node Version:** 14.17.3
* **npm Version:** 6.14.13
* **Operating System:** MacOS 11.4 (20F71)

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

I am using `@typescript-eslint/parser`, I have also tried the `'default'` for which it seems the issue is not present, but I'm not sure.

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "extends": [
      "react-app",
      "react-app/jest",
      "@roydukkey/eslint-config"
    ],
    "rules": {
      "react/jsx-boolean-value": "error",
      "react/jsx-closing-bracket-location": "error",
      "react/jsx-closing-tag-location": "error",
      "react/jsx-child-element-spacing": "error",
      "react/jsx-curly-brace-presence": [
        "error",
        "never"
      ],
      "react/jsx-curly-newline": [
        "error",
        {
          "multiline": "require",
          "singleline": "forbid"
        }
      ],
      "react/jsx-curly-spacing": "error",
      "react/jsx-equals-spacing": [
        "error",
        "never"
      ],
      "react/jsx-filename-extension": [
        "error",
        {
          "extensions": [
            ".jsx",
            ".tsx"
          ]
        }
      ],
      "react/jsx-fragments": [
        "error",
        "element"
      ],
      "react/jsx-indent-props": [
        "error",
        2
      ],
      "react/jsx-indent": [
        "error",
        2
      ],
      "react/jsx-max-props-per-line": [
        "error",
        {
          "maximum": 1,
          "when": "multiline"
        }
      ],
      "react/jsx-no-comment-textnodes": "error",
      "react/jsx-no-duplicate-props": "error",
      "react/jsx-no-undef": "error",
      "react/jsx-no-useless-fragment": "error",
      "react/jsx-props-no-multi-spaces": "error",
      "react/jsx-tag-spacing": "error",
      "react/jsx-wrap-multilines": "error",
      "react/no-unescaped-entities": "error",
      "react/self-closing-comp": "error",
      "react/void-dom-elements-no-children": "error",
      "@typescript-eslint/naming-convention": [
        "error",
        {
          "selector": "default",
          "format": [
            "camelCase"
          ],
          "filter": {
            "regex": "^(Component|Page|Section|Service)$",
            "match": false
          }
        },
        {
          "selector": "default",
          "modifiers": [
            "unused"
          ],
          "filter": {
            "regex": "^(Component|Page|Section|Service)$",
            "match": false
          },
          "format": [
            "camelCase"
          ],
          "leadingUnderscore": "require"
        },
        {
          "selector": "typeLike",
          "format": [
            "PascalCase"
          ]
        },
        {
          "selector": "property",
          "modifiers": [
            "private"
          ],
          "format": [
            "camelCase"
          ],
          "leadingUnderscore": "require"
        },
        {
          "selector": "interface",
          "format": [
            "PascalCase"
          ],
          "custom": {
            "regex": "^I[A-Z]",
            "match": false
          }
        }
      ],
      "@typescript-eslint/no-extra-parens": "off"
    },
    "overrides": [
      {
        "files": [
          "*.d.ts"
        ],
        "rules": {
          "@typescript-eslint/naming-convention": "off"
        }
      }
    ]
  }
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
export default interface AnyInterface {

  get length (): number;

}
```

```bash
eslint --ext js,jsx,ts,tsx ./src
```

**What did you expect to happen?**
> Disallowing usage of specific global variables can be useful if you want to allow a set of global variables by enabling an environment, but still want to disallow some of those. [ref](https://eslint.org/docs/rules/no-restricted-globals)

No issue for `length` getter. Doesn't make sense since length here is not even a variable.

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

`Unexpected use of 'length'. eslint(no-restricted-globals)`

**Are you willing to submit a pull request to fix this bug?** No

**Comment by mdjermanovic** [comment]
Hi @roydukkey, thanks for the issue!

> No issue for `length` getter. Doesn't make sense since length here is not even a variable.

This seems to be a problem in typescript-eslint's scope analysis, which is provided by `@typescript-eslint/parser`. Can you please open an issue in https://github.com/typescript-eslint/typescript-eslint repo?

I'm closing this issue since it isn't actionable in core ESLint.
